### DISCUSSION TITLE
[Help needed] Slow download results - speedtest++ and ookla package showing good results.

### DISCUSSION THREAD

**Discussion by hasechris** [alexjustesen/speedtest-tracker#1949] [Q&A]
Hi,

I was really happy to find your project via the selfhost newsletter. Tried it, was really happy with the webui -  but the download speeds are not good. I get mostly around 100mbit/s in tests, fluctuating around 85 Mbit/s to 98 Mbit/s. One testserver could test my full bandwidth, server ID 39047 "level66.network UG".

Test 1 with server ID 39047 "level66.network UG"
[speedtest.net](https://www.speedtest.net/result/c/05ac5f2d-fb83-49d3-b0d9-dcbdf77d308e)

Test 2 with server ID 39047 "level66.network UG"
[speedtest.net](https://www.speedtest.net/result/c/0ae8b026-6c2c-411d-be26-ce9bd3dd81a7)

this is a repeatable result with this server id with the webui. Other servers only get the mentioned ~ 100 mbit/s.

Test with server ID 18667 "meerfarbig GmbH & Co. KG"
[speedtest.net result](https://www.speedtest.net/result/c/55dd7409-3d0f-4780-a82d-f9cce5932d42)

Test with server ID 44081 "23M GmbH"
[speedtest.net result](https://www.speedtest.net/result/c/ff51d45a-018f-48e5-aa59-7cf1aa75ca51)

Test with server ID 40094 "PVDataNet"
[speedtest.net result](https://www.speedtest.net/result/c/a63e3023-066b-4596-acc3-9e2331b4b2cd)

Test with server ID 54504 "Deutsche Glasfaser"
[speedtest.net result](https://www.speedtest.net/result/c/19bb7560-4ace-42c2-97c0-c6fd4085b9a9)

----

After this I tested directly with the speedtest-cli - but found that I cant use the same server IDs with this package. But with speedtest-cli I also could only get around 100Mbit/s. 

Then I found out that this debian package is not the official package from ookla, so I tried their package via their own package repo. Instantly speedtests with the full 500/50 result instead of the 100Mbit/s results.

Test with server ID 54504 "Deutsche Glasfaser" - now with ookla speedtest package. I forced this server ID because with my browser on my desktop I got around 100Mbit/s with a "single connection" test and this server was tried.
[speedtest.net result](https://www.speedtest.net/result/c/668949f2-177e-44c9-a9db-9f8537791211)

Test with server ID 39873 "Pingless.com - Frankfurt am Main" - with ookla speedtest package
[speedtest.net result](https://www.speedtest.net/result/c/bbf707cd-eb22-4537-9e79-23181b5374c1)

So... I dont know what to make of these results. These different tests were all made on the same system with the same network connection.

Did someone else encounter this problem with this docker image or with the speedtest-cli package?

Greetings from Germany
hasechris

**Comment by alexjustesen** [comment]
Just to confirm on your laptop regardless of server you're getting the speeds you'd expect but on your server where Speedtest Tracker is installed your not?

If so this limit sounds like either a failed ethernet cable or a limited eth port that is only supporting 100mbit speeds.

**Reply by hasechris** [reply]
Hi :-)

> Just to confirm on your laptop regardless of server you're getting the speeds you'd expect but on your server where Speedtest Tracker is installed your not?
> If so this limit sounds like either a failed ethernet cable or a limited eth port that is only supporting 100mbit speeds.

I can get full bandwidth tests **both on my laptop via cable** *but also on the same server*. Also I get good results on the mentioned server here and there. Look at this graph:

![Screenshot 2024-12-23 at 08-28-21 Dashboard - Speedtest Tracker](https://github.com/user-attachments/assets/7fca58bf-5929-499b-815a-dc777b299c41)

As you can clearly see, here and there a testserver seems to be used who can test my full bandwidth.

As mentioned in my first post I can get full results, but only on a fraction of my tests and I think that is because of speedtest-cli. 

If I use the official ookla package **on the same server** as the speedtest-tracker container I can get matching results (to my bandwidth) on every single speedtest.

**Reply by hasechris** [reply]
additionally here are my speedtest-tracker results (If you want I can also make an export via the webui)

![image](https://github.com/user-attachments/assets/73332cad-a5d1-466e-a491-2610fb30373b)

**Reply by alexjustesen** [reply]
Running tests this frequently we've seen people getting throttled. @svenvg93 and I _think_ the limit is roughly 30 calls/hour through the CLI.

Try testing less frequently like every 15min and see if these stabilize.

**Reply by svenvg93** [reply]
To add; Looking at the results it seems some problem with these specific servers. Might capacities on the server or transit/peering between your isp and their networks. As some do have good results.

**Reply by hasechris** [reply]
> Running tests this frequently we've seen people getting throttled. @svenvg93 and I _think_ the limit is roughly 30 calls/hour through the CLI.
> 
> Try testing less frequently like every 15min and see if these stabilize.

Yeah, I quickly found that one out. Getting a 403 Forbidden on speedtest was puzzling for a bit. I changed my schedule to hourly, but the result did not change. Most of my tests are limited around 100 mbit/s, but some get the full bandwidth.

![image](https://github.com/user-attachments/assets/604a6d83-1ce7-4be7-97ea-6871dec7bb5f)
![image](https://github.com/user-attachments/assets/fa7528df-1f17-4ec9-a3db-7aa653f6cda4)

Would it be possible for you to build an image for tests with the official ookla package? With that package I could make speedtests every 10 minutes for over 2 hours without a problem and get full 500/50 results.

**Reply by svenvg93** [reply]
We are using the official Ookla package.
we call the `speedtest` command like you do on every other system using the package. And collect the results.

I still think the limitation is on the server/network (isp) side and not on the application. 

I had the same problem I the past where some servers where always giving a low result compared to others. 

You can set the servers to use with the `SPEEDTEST_SCHEDULE` [variable](https://docs.speedtest-tracker.dev/getting-started/environment-variables). This way you test only against server that you know are able to give you a good result.

**Reply by hasechris** [reply]
> We are using the official Ookla package. we call the `speedtest` command like you do on every other system using the package. And collect the results.
> 
> I still think the limitation is on the server/network (isp) side and not on the application.
> 
> I had the same problem I the past where some servers where always giving a low result compared to others.
> 
> You can set the servers to use with the `SPEEDTEST_SCHEDULE` [variable](https://docs.speedtest-tracker.dev/getting-started/environment-variables). This way you test only against server that you know are able to give you a good result.

so i tested that now. I specified the two test servers which got me full bandwidth and tested around every 15 Minutes (a bit spaced out). I really don't like to specify speedtest servers because the specified targets could have a problem, would be offline or just slow.

```
      - SPEEDTEST_SCHEDULE="4,17,35,48 * * * *"
      - SPEEDTEST_SERVERS=37492,39873
```

This is the result when only using these two servers:
![image](https://github.com/user-attachments/assets/0b896faf-12a0-4224-9fdb-d61cd9289f6c)

So i clearly don't have a problem with my internet connection or the server on which this speedtest container is running.

Does someone know or understand why the speedtest-cli picks so many servers which clearly cant test my full bandwidth?

Greetings
hasechris

**Reply by alexjustesen** [reply]
> Does someone know or understand why the speedtest-cli picks so many servers which clearly cant test my full bandwidth?

As best as I can tell it'll pick a server based on proximity/ping and not based on download/upload available bandwidth. From what I've also seen this problem is more prevalent in EU servers and could have something to do with more restrictive bandwidth policies.

**Reply by yeathn** [reply]
Im also from Europe and had the same issue with multiple other servers.

After choosing 37492, 39873 i got full speed as well. Thx :)
